src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -317,6 +335,254 @@ def on_time_step__prepare(self, event: Event):
         event : vivarium.framework.event.Event
 
         """
-        population = self.population_view.subview(["age"]).get(event.index, query="alive == 'alive'")
+        population = self.population_view.subview(["age"]).get(
+            event.index, query="alive == 'alive'"
+        )
         population["age"] += to_years(event.step_size)
         self.population_view.update(population)
+
+    def assign_gen_pop_guardians(self, pop: pd.DataFrame) -> pd.DataFrame:
+        """
+
+        Parameters
+        ----------
+        pop: State stable of simulants
+
+        Returns
+        -------
+        pd.Dataframe that will be pop with an additional guardians column containing the index for that simulant's
+          guardian..
+        """
+
+        # Initialize column
+        pop["guardian_1"] = data_values.UNKNOWN_GUARDIAN_IDX
+        pop["guardian_2"] = data_values.UNKNOWN_GUARDIAN_IDX
+        # Helper lists
+        non_relatives = ["Roommate", "Other nonrelative"]
+        children = ["Biological child", "Adopted child", "Foster child", "Stepchild"]
+        parents = ["Parent", "Parent-in-law"]
+        partners = [
+            "Opp-sex spouse",
+            "Opp-sex partner",
+            "Same-sex spouse",
+            "Same-sex partner",
+        ]
+
+        # Get household structure for population to vectorize choices
+        # Non-GQ population
+        gen_population = pop.loc[~pop["household_id"].isin(data_values.GQ_HOUSING_TYPE_MAP)]
+        household_structure = self.get_household_structure(gen_population)
+
+        # Children helper index groups
+        children_of_rp_idx = household_structure.loc[
```

Comment:
instead of `rp` could you do something like `ref_person`?

Reply:
or include a comment here like #children of reference person index groups so that reading thru the rest i know `rp` is reference person

Reply:
^ after seeing how many times you used `rp` im maybe leaning more towards just defining it here / near the top before using it a bunch -- i guess depending on the degree to which actually renaming it would make the lines long/unwieldy